pbm/config.go

```diff
@@ -82,13 +92,18 @@ func (p *PBM) SetConfigByte(buf []byte) error {
 }
 
 func (p *PBM) SetConfig(cfg Config) error {
-	_, err := p.Conn.Database(DB).Collection(ConfigCollection).UpdateOne(
+	err := cfg.Storage.Cast()
+	if err != nil {
+		return errors.Wrap(err, "cast storage")
+	}
+
+	_, err = p.Conn.Database(DB).Collection(ConfigCollection).UpdateOne(
 		p.ctx,
 		bson.D{},
 		bson.M{"$set": cfg},
-		options.Update().SetUpsert(true))
-
-	return err
+		options.Update().SetUpsert(true),
+	)
+	return errors.Wrap(err, "mongo UpdateOne")
```

Comment:
How about this for providing some context that will make the error string more unique?, if it shows up in JIRA tickets etc.
```suggestion
	return errors.Wrap(err, "mongo ConfigCollection UpdateOne")
```